### PR TITLE
v1.4 fix issues

### PR DESCRIPTION
**Pull request by bklimey**
Update Find command message to make it clearer that social handle only allows to search userid and not social platform
image

Remove ability to sort Tag and SocialHandle as it may not be meaningful.
image

Make it explicit in message usage of optional fields that leaving the field blank will remove that field.
image

Other fixes:
- Move `ImageStorage` from `storage` to `commons` package and rename it to `ImageUtil`
  - Reason: `ImageStorage` does not fit the interface of `storage`. Therefore, it is moved away from `storage` to avoid confusion.
- Fix multiplicity of `SocialHandle` in diagram
  - Previously, it was `0..1`. Therefore, it is changed to `*`
- Standardize fields with space to use underscore. 
  - Reason: There is a mix of fields using spaces and underscore. Underscore is chosen to avoid confusion where people mistake a single field for 2 fields (e.g. `stat TUTORIAL GROUP`)

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Nationality.java [hunk 1]
```diff
@@ -17,7 +17,8 @@
 public class Nationality {
 
     public static final List<String> VALID_NATIONALITIES = readValidNationalities();
-    public static final String MESSAGE_CONSTRAINTS = "Nationality can take any values, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Nationality can be any valid nationality spelled in full."
+            + "Leaving it blank will remove the Nationality field. \n";
```
Comment (kevinmingtarja) [inline src/main/java/seedu/address/model/person/Nationality.java:21]: Should we add space after this full stop?

File: src/main/java/seedu/address/model/person/Phone.java [hunk 2]
```diff
@@ -12,7 +12,8 @@
 
     public static final String MESSAGE_CONSTRAINTS =
             "Phone numbers should only contain numbers and optionally start with a  '+'. "
-            + "Phone numbers should be at least 3 digits and at most 15 digits long.";
+            + "Phone numbers should be at least 3 digits and at most 15 digits long."
+            + "Leaving it blank will remove the Phone Number field. \n";
```
Comment (kevinmingtarja) [inline src/main/java/seedu/address/model/person/Phone.java:16]: Here as well

**Review by kevinmingtarja — APPROVED** [review]
LGTM!

**Review by hsiaotingluv — APPROVED** [review]
No issue but do resolve the merge conflict first after CY's PR on error messages is merged!

**Pull request merged**